openwillis/measures/text/speech_attribute.py

```diff
@@ -4,154 +4,404 @@
+    phrases_idxs: list
+        A list of tuples containing
+            the start and end indices of the phrases in the JSON object.
+    phrases: list
+        A list of phrases extracted from the JSON object.
 
     Returns:
     ...........
-    bool: True if the json response object is from Amazon Transcribe, False otherwise.
+    turns_idxs: list
+        A list of tuples containing
+            the start and end indices of the turns in the JSON object.
+    turns: list
+        A list of turns extracted from the JSON object.
+    phrases_idxs: list
+        A list of tuples containing
+            the start and end indices of the phrases in the JSON object.
+    phrases: list
+        A list of phrases extracted from the JSON object.
 
     ------------------------------------------------------------------------------------------------------
     """
-    return 'jobName' in json_conf and 'results' in json_conf
 
-def filter_transcribe(json_conf, speaker_label=None):
+    # phrase-split for the speaker label
+    phrases_idxs2 = []
+    phrases2 = []
+    for i, phrase in enumerate(phrases_idxs):
+        start_idx = phrase[0]
+        if item_data[start_idx].get("speaker_label", "") == speaker_label:
+            phrases_idxs2.append(phrase)
+            phrases2.append(phrases[i])
+
+    phrases_idxs = phrases_idxs2
+    phrases = phrases2
+
+    # turn-split for the speaker label
+    start_idx = 0
+    for i, item in enumerate(item_data):
+        if (
+            i > 0
+            and item.get("speaker_label", "") == speaker_label
+            and item_data[i - 1].get("speaker_label", "") != speaker_label
+        ):
+            start_idx = i
+        elif (
+            i > 0
+            and item.get("speaker_label", "") != speaker_label
+            and item_data[i - 1].get("speaker_label", "") == speaker_label
+        ):
+            turns_idxs.append((start_idx, i - 1))
+            # create turns texts
+            turns.append(
+                " ".join(
+                    [
+                        item["alternatives"][0]["content"]
+                        for item in item_data[start_idx:i]
+                    ]
+                )
+            )
+
+    if start_idx not in [item[0] for item in turns_idxs]:
+        turns_idxs.append((start_idx, len(item_data) - 1))
+        turns.append(
+            " ".join(
+                [
+                    item["alternatives"][0]["content"]
+                    for item in item_data[start_idx:]
+                ]
+            )
+        )
+
+    return turns_idxs, turns, phrases_idxs, phrases
+
+
+def filter_transcribe(json_conf, measures, speaker_label=None):
     """
     ------------------------------------------------------------------------------------------------------
 
-    This function extracts the text and filters the JSON data for Amazon Transcribe json response objects.
+    This function extracts the text and filters the JSON data
```

Comment:
same suggestion as above function:
- manage exception handling
- split it into smaller function
- try to include helper function in util file